### PR TITLE
[NIC-30] Add update document status and correct save

### PR DESCRIPTION
**Pull request by AksyonovaKaterina**

### CODE REVIEW

File: backend/document-service/src/main/resources/http/documentControllerRequests.http [hunk 1]
```diff
@@ -20,6 +18,17 @@ Authorization: Bearer <access_token>
 Content-Type: application/json
 
 ### Get all document
-GET http://localhost:8083/public/api/v1/document/all?sort=type,desc
+GET http://localhost:8083/public/api/v1/document/all?sort=orderDate,desc
+Authorization: Bearer <access_token>
+Content-Type: application/json
+
+### Get document
```
Comment (V7074) [inline backend/document-service/src/main/resources/http/documentControllerRequests.http:25]:
```suggestion
### Update document
```

File: backend/document-service/src/main/resources/http/documentControllerRequests.http [hunk 2]
```diff
@@ -20,6 +18,17 @@ Authorization: Bearer <access_token>
...
+PUT http://localhost:8083/public/api/v1/document/638759f44da9805de8877486
 Authorization: Bearer <access_token>
-Content-Type: application/json
\ No newline at end of file
+Content-Type: application/json
+
+{
+  "orderDate": "2022-10-23",
+  "type": "WORK_STATEMENT",
+  "status": "IN_PROGRESS"
```
Comment (V7074) [inline backend/document-service/src/main/resources/http/documentControllerRequests.http:33]: I think we should pass only the `status` field here

File: backend/document-service/src/main/kotlin/com/nosql/document/dto/DocumentDto.kt [hunk 3]
```diff
@@ -4,6 +4,7 @@ import com.fasterxml.jackson.annotation.JsonFormat
 import com.fasterxml.jackson.annotation.JsonProperty
 import com.nosql.document.constants.jackson.DATE_PATTERN
 import com.nosql.document.enumerator.DocumentStatus
+import com.nosql.document.enumerator.DocumentStatus.ORDERED
 import com.nosql.document.enumerator.DocumentType
 import java.time.Instant
 import java.util.Date
```
Comment (V7074) [inline backend/document-service/src/main/kotlin/com/nosql/document/dto/DocumentDto.kt:10]: make this `DocumentDto` as `data class`

**Review by V7074 — CHANGES_REQUESTED** [review]

File: backend/document-service/src/main/kotlin/com/nosql/document/controller/DocumentController.kt [hunk 4]
```diff
@@ -72,6 +73,16 @@ class DocumentController(
         @Valid @RequestBody documentDto: DocumentDto,
     ) = documentService.save(userAuthInfo.userId, documentDto)
 
+    @PutMapping(
+        value = [ID_URL_PATH],
+        consumes = [APPLICATION_JSON_VALUE],
+        produces = [APPLICATION_JSON_VALUE],
+    )
```
Comment (V7074) [inline backend/document-service/src/main/kotlin/com/nosql/document/controller/DocumentController.kt:80]: I think this is `PATCH` - We are updating not whole object, but only few fields

**Review by V7074 — CHANGES_REQUESTED** [review]

File: backend/document-service/src/main/resources/http/documentControllerRequests.http (hunk 2 above)
Comment (AksyonovaKaterina) [reply backend/document-service/src/main/resources/http/documentControllerRequests.http:33]: fixed

File: backend/document-service/src/main/resources/http/documentControllerRequests.http (hunk 1 above)
Comment (AksyonovaKaterina) [reply backend/document-service/src/main/resources/http/documentControllerRequests.http:25]: fixed

File: backend/document-service/src/main/kotlin/com/nosql/document/dto/DocumentDto.kt (hunk 3 above)
Comment (AksyonovaKaterina) [reply backend/document-service/src/main/kotlin/com/nosql/document/dto/DocumentDto.kt:10]: It make no sense, we do not user toString

File: backend/document-service/src/main/kotlin/com/nosql/document/controller/DocumentController.kt (hunk 4 above)
Comment (AksyonovaKaterina) [reply backend/document-service/src/main/kotlin/com/nosql/document/controller/DocumentController.kt:80]: fixed

**Pull request closed without merging**